java/documentation/library-coverage/frameworks.csv

```diff
@@ -1,8 +1,10 @@
-Framework name,URL,Package prefix
+Framework name,URL,Package prefixes
 Java Standard Library,,java.*
-Google,,com.google.common.*
-Apache,,org.apache.*
+Java extensions,,javax.*
+Google Guava,https://guava.dev/,com.google.common.*
 Apache Commons IO,https://commons.apache.org/proper/commons-io/,org.apache.commons.io
+Apache Commons Lang,https://commons.apache.org/proper/commons-lang/,org.apache.commons.lang3
+Apache Commons Text,https://commons.apache.org/proper/commons-text/,org.apache.commons.text
+Apache HttpComponents,https://hc.apache.org/,org.apache.hc.core5.* org.apache.http
```

Comment:
Some of the packages are specified using exact names (`org.apache.http`) and others using patterns/prefixes (`org.apache.hc.core5.*`). I note that this matches the current grouping of the coverage query output in `java/documentation/library-coverage/coverage.csv` for these packages.

Have you checked what happens if the grouping of the query output changes (as a result of adding/removing models or changing the coverage query)? For example, if the `org.apache.http` package is split into subpackages, or subpackages of `org.apache.hc.core5.*` are combined into a parent package? Would all the numbers still add up correctly regardless of how the various packages/patterns are specified?

Reply:
@yo-h I'm not sure I fully understand the question. There are two ways of specifying package names: (i) exact match, (ii) prefix match when the last character is `*`. 

If a row is returned from the QL query with `org.apache.http.abc` package ID, that will NOT be included in the currently defined `Apache HttpComponents` framework, because it doesn't match the `org.apache.hc.core5.` prefix and it's not equal to `org.apache.http`. Maybe this is not the desired behaviour, should I change the `org.apache.http` to `org.apache.http*` (note, `*` at the end, and not `.*`)? `org.apache.http.abc` would be included in the `Others` row.

Also, if QL returns `org.apache.hc.core5` that wouldn't be matched currently by `org.apache.hc.core5.*`. I should probably change the latter to `org.apache.hc.core5*` (note, no `.` before the `*`).

Is this what you meant?

Reply:
Thanks for clarifying the semantics. On further thought, I think it's fine as it is for now. If the matching were to change, we will get a PR comment thanks to your other work, and can make adjustments then.

For future reference, would `org.apache.http*` match `org.apache.https`? (I don't think we want that to match, but instead we can specify both `org.apache.http` and `org.apache.http.*` where necessary.)

Reply:
Yes, `org.apache.http*` would match `org.apache.https`, if that's undesired than specifying both `org.apache.http` and `org.apache.http.*` is the solution.